python/seldon_core/microservice.py

```diff
@@ -351,107 +333,108 @@ def main():
     else:
         user_object = user_class(**parameters)
 
-    port = args.port
+    http_port = args.http_port
+    grpc_port = args.grpc_port
     metrics_port = args.metrics_port
 
-    if args.tracing:
-        tracer = setup_tracing(args.interface_name)
+    # if args.tracing:
+    #    tracer = setup_tracing(args.interface_name)
+
+    seldon_metrics = SeldonMetrics(worker_id_func=os.getpid)
+    # TODO why 2 ways to create metrics server
```

Comment:
@RafalSkolasinski can probably provide more details, but I imagine this is due to how Flask separates requests by threads vs Gunicorn, who separates requests by process? Although TBH Gunicorn also separates requests by threads within each process...

Reply:
Hmm I do remember that gunicord and grpc had repsective quirks that ended up requiring adding this, as I asked the same question. Probably one for Rafal as I remember he explained the reasoning, it seems it's actually required in this case

Reply:
Oh yeah, this was quite crucial one. It was related to multi-threading / multi-processing. 
I was using different methods to identify workers: process's id and thread's names.

I have the feeling that I should just come with a method that will just combine `pid` and `thread name` into a single string that identifies workers. This will work then for both threading / multiprocessing case -> especially that now in production setup one can set gunicorn workers and threads independently.

Reply:
On more details, this is used in two places:
- internally to track data from each worker independently -> so output from `worker_id_func` must be hashable
- in tags exposed to prometheus as `worker_id` -> so must be cast able to `string`


Something like
```
        seldon_metrics = SeldonMetrics(	
            worker_id_func=lambda: f"{threading.current_thread().name}-{os.getpid()}"
        )
```
probably would do the job nicely

Reply:
^ okay, not so sure about above, investigating now...

Reply:
Looked a bit more into it, I think now  that using combined pid and thread name seems the most ideal here. 

Have a commit on top of @cliveseldon's branch that's changing it: https://github.com/RafalSkolasinski/seldon-core/commit/7d0d994b9634675dec8d9097ffa7f6e46b07578c

Can include in this PR or create a separate PR to merge later.

Reply:
If you can merge into this PR or yes let's create a new one post this

Reply:
Can go as separate PR, that may be easier.